### PR TITLE
Feat: Create CollapsedMenu Component

### PR DESCRIPTION
**Pull request by neli-rashkova**
## Description
Created `CollapsedMenu` component that consists of a caption and a sublist. Clicking on the trigger content will display the collapsed menu.
Props:

- triggerContent: the component passed in that will be the trigger content.
- caption: string
- subList: an array of objects, each object has two properties, text and onClick

## Link to JIRA ticket
[Link to JIRA ticket](https://implentio.atlassian.net/browse/IUT-29?atlOrigin=eyJpIjoiMjkyNTRlZWI0M2NkNGEzNmIxMTU1ZDU3ZGZhZjdkYWYiLCJwIjoiaiJ9)

## Applicable Screenshots
**Note: Icon component is used as an example trigger content to click on and show how collpased menu component renders**
Screenshot 2024-02-09 at 10 46 01 AM
Screenshot 2024-02-09 at 10 46 20 AM
Screenshot 2024-02-09 at 10 46 29 AM
Screenshot 2024-02-09 at 10 46 36 AM

## Testing Instructions
- Fetch and checkout my branch `git fetch && git checkout feat/IUT-29/CollapsedMenu`
- `npm run storybook`
- Check to see that when clicking on trigger content, the collapsed menu is displayed. Hover over sublist.

## Checklist:
Check what is applicable by putting an 'x' in between the []:
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have tested my code thoroughly.
- [ ] I made documentation changes (optional).

### CODE REVIEW

**Review by launchupio — CHANGES_REQUESTED** [review]
@neli-rashkova sorry for adding in additional design work here, but the hover state has actually now been specified by design team. Can you take al ook at this and change the designs accordingly and ping me again after when PR is ready?

Appreciate it!

https://www.figma.com/file/UYBp8YXmCgUqhiOsrkHd6V/Implentio-Library?type=design&node-id=182-13759&mode=design&t=x0pNTA0argGLEzaQ-0

File: src/components/CollapsedMenu/CollapsedMenu.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+export const IntelligenceCollapsedMenu: Story = {
+    args: {
+        triggerContent: exampleTriggerContent(),
+        caption: 'Intelligence',
+        subList: [ 
+            { 
+                text: "Reports",
+                onClick: () => console.log("Test")
```
Comment (launchupio) [inline src/components/CollapsedMenu/CollapsedMenu.stories.tsx:53]: all of these should use storybook actions instead: e.g. `onClick: action('reportsClick')`

File: src/components/CollapsedMenu/CollapsedMenu.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            }, 
+            { 
+                text: "Analytics",
+                onClick: () => console.log("Testing")
```
Comment (launchupio) [inline src/components/CollapsedMenu/CollapsedMenu.stories.tsx:57]: all of these should use storybook actions instead

File: src/components/CollapsedMenu/CollapsedMenu.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+export const ReconciliationsCollapsedMenu: Story = {
+    args: {
+        triggerContent: exampleTriggerContent(),
+        caption: 'Reconciliations',
+        subList: [ 
+            { 
+                text: "Invoices",
+                onClick: () => console.log("Test")
```
Comment (launchupio) [inline src/components/CollapsedMenu/CollapsedMenu.stories.tsx:29]: all of these should use storybook actions instead

File: src/components/CollapsedMenu/CollapsedMenu.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            }, 
+            { 
+                text: "Credit Tracker",
+                onClick: () => console.log("Testing")
```
Comment (launchupio) [inline src/components/CollapsedMenu/CollapsedMenu.stories.tsx:33]: all of these should use storybook actions instead (name them better too)

File: src/components/CollapsedMenu/CollapsedMenu.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import type { Meta, StoryObj } from "@storybook/react";
+import { CollapsedMenu } from './CollapsedMenu';
+import { IconButton } from '../IconButton/IconButton';
+import { SublistItem } from './CollapsedMenu';
+
```
Comment (launchupio) [inline src/components/CollapsedMenu/CollapsedMenu.stories.tsx:6]: import storybook actions

File: src/components/CollapsedMenu/CollapsedMenu.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    tags: ["autodocs"],
+};
+
+export default meta;
+type Story = StoryObj<typeof CollapsedMenu>;
+
+// The function below renders the icon component which is used as an example to help show how collapsed menu works and renders on the screen.
+const exampleTriggerContent = () => {
+    return (
+        <IconButton icon={'MagnifyingGlassIcon'} size='medium' onClick={() => console.log('Hello')} />
```
Comment (launchupio) [inline src/components/CollapsedMenu/CollapsedMenu.stories.tsx:18]: use storybook action for onClicks

File: src/components/CollapsedMenu/CollapsedMenu.stories.tsx (hunk 1 above)
Comment (neli-rashkova) [reply src/components/CollapsedMenu/CollapsedMenu.stories.tsx]: fixed

File: src/components/CollapsedMenu/CollapsedMenu.stories.tsx (hunk 2 above)
Comment (neli-rashkova) [reply src/components/CollapsedMenu/CollapsedMenu.stories.tsx]: fixed

File: src/components/CollapsedMenu/CollapsedMenu.stories.tsx (hunk 3 above)
Comment (neli-rashkova) [reply src/components/CollapsedMenu/CollapsedMenu.stories.tsx]: fixed

File: src/components/CollapsedMenu/CollapsedMenu.stories.tsx (hunk 4 above)
Comment (neli-rashkova) [reply src/components/CollapsedMenu/CollapsedMenu.stories.tsx]: fixed

File: src/components/CollapsedMenu/CollapsedMenu.stories.tsx (hunk 5 above)
Comment (neli-rashkova) [reply src/components/CollapsedMenu/CollapsedMenu.stories.tsx:6]: ok!

File: src/components/CollapsedMenu/CollapsedMenu.stories.tsx (hunk 6 above)
Comment (neli-rashkova) [reply src/components/CollapsedMenu/CollapsedMenu.stories.tsx]: ok!

**Review by joseph-hc-jun — APPROVED** [review]
Looks good! Thank you :)

**Review by launchupio — APPROVED** [review]
🚀

**Pull request merged**